mungegithub/mungers/approvers/owners.go

```diff
@@ -0,0 +1,248 @@
+	unapproved := sets.NewString()
+	for fn := range o.GetOwnersSet() {
+		approvers := o.repo.Approvers(fn)
+		if approvers.Intersection(currentlyApproved).Len() == 0 {
+			unapproved.Insert(fn)
+		}
+	}
+	return unapproved
+}
+
+// Shuffles the potential approvers so that we don't always suggest the same people
+func (o Owners) GetShuffledApprovers() []string {
+	approversList := o.GetAllPotentialApprovers()
+	order := rand.New(rand.NewSource(o.seed)).Perm(len(approversList))
+	people := make([]string, 0, len(approversList))
+	for _, i := range order {
+		people = append(people, approversList[i])
+	}
+	return people
+}
+
+// removeSubdirs takes a list of directories as an input and returns a set of directories with all
+// subdirectories removed.  E.g. [/a,/a/b/c,/d/e,/d/e/f] -> [/a, /d/e]
+func removeSubdirs(dirList []string) sets.String {
+	toDel := sets.String{}
+	for i := 0; i < len(dirList)-1; i++ {
+		for j := i + 1; j < len(dirList); j++ {
+			// ex /a/b has prefix /a so if remove /a/b since its already covered
+			if strings.HasPrefix(dirList[i], dirList[j]) {
+				toDel.Insert(dirList[i])
+			} else if strings.HasPrefix(dirList[j], dirList[i]) {
+				toDel.Insert(dirList[j])
+			}
+		}
+	}
+	finalSet := sets.NewString(dirList...)
+	finalSet.Delete(toDel.List()...)
+	return finalSet
+}
+
+// getMessage returns the comment body that we want the approval-handler to display on PRs
+// The comment shows:
+// 	- a list of approvers files (and links) needed to get the PR approved
+// 	- a list of approvers files with strikethroughs that already have an approver's approval
+// 	- a suggested list of people from each OWNERS files that can fully approve the PR
+// 	- how an approver can indicate their approval
+// 	- how an approver can cancel their approval
+// THIS IS STILL UGLY AND NEEDS TO BE FIXED
+func getMessage(o Owners, alreadyApproved sets.String, org, project string) string {
+	context := bytes.NewBufferString("")
+	if len(alreadyApproved) != 0 {
+		context.WriteString("The following people have approved this PR: ")
+		context.WriteString("*" + strings.Join(alreadyApproved.List(), ", ") + "*\n\n")
+	}
+
+	context.WriteString("Needs approval from an approver in each of these OWNERS Files:\n")
+	unapprovedFiles := o.UnapprovedFiles(alreadyApproved)
+	for _, path := range o.GetOwnersSet().List() {
+		fullOwnersPath := filepath.Join(path, ownersFileName)
+		approverSet := o.repo.Approvers(path).Intersection(alreadyApproved)
+		link := fmt.Sprintf("https://github.com/%s/%s/blob/master/%v", org, project, fullOwnersPath)
+		if unapprovedFiles.Has(path) {
+			context.WriteString(fmt.Sprintf("- **[%s](%s)** \n", fullOwnersPath, link))
+		} else {
+			context.WriteString(fmt.Sprintf("- ~~[%s](%s)~~ [%v]\n", fullOwnersPath, link, strings.Join(approverSet.List(), ",")))
+		}
+	}
+	context.WriteString("\n")
+	suggestedApprovers := o.GetSuggestedApprovers()
+	toBeAssigned := sets.NewString(suggestedApprovers...).Difference(alreadyApproved)
```

Comment:
Function for getting people that should be assigned or is it okay to have the logic here?